utils/hardware_info.go

```diff
@@ -0,0 +1,15 @@
+package utils
+
+import (
+	"runtime"
+)
+
+// GetArchitectureInfo returns the architecture of the board.
+func GetArchitectureInfo() string {
+	return runtime.GOARCH
+}
+
+// GetOSInfo returns the OS of the board.
+func GetOSInfo() string {
```

Comment:
This function appears to be unused. Is it supposed to be used somewhere? (and same suggestion about not having the function at all and just using `runtime.GOOS` wherever this was supposed to be used.)